extensions/iam/decentralized-identity/registration-service/src/main/java/org/eclipse/dataspaceconnector/iam/registrationservice/crawler/CrawlerJob.java

```diff
@@ -38,12 +38,18 @@
 
 import static java.lang.String.format;
 
-// has to be "public", otherwise quartz won't be able to access is
 public class CrawlerJob implements Job {
 
     private static final String DIDS_PATH = "dids";
+    private static final Clock CLOCK = Clock.systemUTC();
```

Comment:
we can add method to `CrawlerContext` builder to add Clock. And then while creating Crawler config(org/eclipse/dataspaceconnector/iam/registrationservice/crawler/CrawlerExtension.java:93) we can add the clock from `ServiceExtensionContext` 

```
var crawlerConfig = CrawlerContext.Builder.create()
               .....
                .clock(context.getClock())
                .build();
```

Reply:
This class is instantiated by Quartz through the no-args constructor

Reply:
Not sure if I understood it correctly because there are other values which crawler job gets from crawler context so in the same way we can have the clock as well.

See below comment for some more details